### PR TITLE
Finish: Feature add reducers and actions.

### PR DESCRIPTION
**Pull request by Zafron047**
Hello Dear reviewer,

I hope this message finds you well. I'm excited to share the progress I've made on Part 2 of this project. Below are the completed tasks:

- [x] Added Redux Toolkit: Installed react-redux and @reduxjs/toolkit using npm.
- [x] Organized Redux Logic: Created a new directory /src/redux to contain all Redux-related files.
- [x] Configured Redux Store: Created a store at /src/redux/store.js using configureStore.

- [x] For the first slice of state (Books):
  - Defined a slice of state for books with an array of books (initially empty) and reducers to add and remove books.
  - For the second slice of state (Categories):

- [x] Created a slice of state for categories with an array of categories (initially empty).
  - Implemented a reducer to check the status and always return "Under construction" (as the initial state).
  - I have used the "feature folder" approach for file structuring and adhered to the Ducks pattern for Redux files, resulting in the following directory structure:

```
/src
|---/components
|---/redux
|--- /books
| booksSlice.js
|--- /categories
| categoriesSlice.js
| store.js
| index.js
```
_NOTE: As per instructions: **No changes in React components are required in this project.** So testing the app will be done in next part of the project_

I am submitting this PR for review and appreciate any feedback or suggestions. Once approved, I'll be ready to move on to Part 3 of the project.

Thank you for your support, and I look forward to your feedback.

Best regards,
@Zafron047

### CODE REVIEW

**Review by Strangeal — COMMENTED** [review]
Hi @Zafron047, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- [x] It appears that your submitted project is missing this essential component. To rectify this, kindly consider the following improvements:
    -  Utilize the [README](https://github.com/microverseinc/readme-template) template provided in the project requirements, This will ensure consistency and completeness in your documentation. 

    - Customize the template to suit your project details. Provide a clear and concise overview of your project, including its description, key features, future features, and any relevant instructions or prerequisites for users. 

    - Consider consuming sections such as installation instructions, usage examples, configuration details, and information about any external dependencies or libraries used. Include information about the project's license, copyright, or any other legal considerations. 
    - Proofread and format your `README` to ensure it is easy to read, well-organized, and free of errors. Remember, documenting your project through a professional README is crucial for other developers to understand, use, and contribute to your project effectively.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1,3 +1,171 @@
...
+
+**I would like to thank microverse for guiding and providing with all great material to do my projects**
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT.md) licensed.
+
 # Getting Started with Create React App
 
 This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (DammyShittu) [inline README.md]: - [x] Kindly remove this line to the end of your readme document. It contains a lot of generic texts generated by React.

**Comment by Zafron047** [comment]
@Strangeal Thanks for the review. I have now 

- [x] added reaadme file.

Regards,
@Zafron047

**Review by DammyShittu — COMMENTED** [review]
Hi @Zafron047,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly remove the highlighted part of the readme with generic texts.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md (hunk 1 above)
Comment (Zafron047) [reply README.md]: @DammyShittu When I was updating the readme file. I kept the react generated texts that were relevant to my project and removed the irrelevant ones. Do I still need to remove texts relevant to my project?

**Review by IngDavidTM — APPROVED** [review]
Hi @Zafron047 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Zafron047** [comment]
@IngDavidTM Thanks for Reviewing the PR.

Regards,
@Zafron047

**Pull request merged**